### PR TITLE
Update Team Attribution to ensure selectedId is set on mounted. Updat…

### PR DESCRIPTION
**Pull request by mcstover**
…e Campaign landing page logic to handle SPA transitions and gracefully handle sessions with either the current promo or some other bonus credit applied.

### CODE REVIEW

File: src/components/Checkout/TeamAttribution.vue [hunk 1]
```diff
@@ -55,6 +55,9 @@ export default {
 			return _orderBy(this.teams, 'name');
 		}
 	},
+	mounted() {
+		this.selectedId = this.teamId || 0;
```
Comment (mcstover) [inline src/components/Checkout/TeamAttribution.vue:59]: no nextTick needed

File: src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue [hunk 2]
```diff
@@ -534,9 +534,9 @@ export default {
 		// check for applied promo
 		this.verifyOrApplyPromotion();
 		// update basket state if any loans are already in the basket
-		if (this.itemsInBasket.length) {
-			this.updateBasketState();
-		}
+		// if (this.itemsInBasket.length) {
+		// 	this.updateBasketState();
+		// }
```
Comment (mcstover) [inline src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue:539]: experimenting with removing this completely. Pretty sure it's a redundant update at this point

File: src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue [hunk 3]
```diff
@@ -711,10 +707,19 @@ export default {
...
+					this.promoData = response.data?.shop?.promoCampaign;
+					this.promoApplied = true;
+					this.loadingPromotion = false;
+					this.promoErrorMessage = null;
+				} else {
+					// Handle response and any potential errors
+					// > this reveals and prior error messages from the promo application
+					this.promoApplied = false;
+					this.loadingPromotion = false;
+				}
```
Comment (mcstover) [inline src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue:721]: Moved the CampaignStatus update here after we've gone through a basket and promo check.

File: src/util/campaignUtils.js [hunk 4]
```diff
@@ -120,6 +120,7 @@ export function getPromoFromBasket(promoFundId, apollo) {
 		query: promoCampaign,
 		variables: {
 			promoFundId: String(promoFundId)
-		}
+		},
+		fetchPolicy: 'network-only',
```
Comment (mcstover) [inline src/util/campaignUtils.js:124]: Ensures we update if there is prior bonus credit found in the initial header query.

**Review by ryan-ludwig — APPROVED** [review]

**Pull request merged**